Tasgrid/tasgridWrapper.hpp

```diff
@@ -181,7 +186,13 @@ class TasgridWrapper{
     void writeMatrix(std::string const &filename, int rows, int cols, const double mat[]) const;
     void printMatrix(int rows, int cols, const double mat[], bool isComplex = false) const;
 
-private:
+    // Overloads.
+    template<typename T>
+    inline void writeMatrix(std::string const &filename, Data2D<T> mat) { writeMatrix(filename, mat.getNumStrips(), mat.getStride(), mat.data()); }
```

Comment:
Ouch, right in the performance!

Pass by reference and check const-correctness. Same for the other overload.